metaci/build/tasks.py

```diff
@@ -229,3 +239,31 @@ def delete_scratch_org(org_instance_id):
         return f"Deleted org instance #{org.id}"
     else:
         return f"Failed to delete org instance #{ord.id}"
+
+
+class BuildSpec(T.NamedTuple):
+    repo_name: str
+    branch_name: str
+    commit: str
+    plan_name: str
+    username_or_email: str
+    timeout: int
+    no_lock: bool = False
+
+
+def run_one_off_build(build):
+    build_config = BuildSpec(
+        repo=build.repo.name,
+        branch_name=build.branch_name,
+        commit=build.commit,
+        plan=build.plan.name,
+        username_or_email=build.org.name,
+        timeout=build.plan.build_timeout,
+    )
+
+    auto_scaler_app_name = list(settings.AUTOSCALERS.values())[1]
```

Comment:
Using AUTOSCALERS for this doesn't make much sense to me, since it's a list of apps that should be scaled based on queue size, which is _not_ what we're doing for the one-off dynos. I think it should be a new setting with the name of the app to use for one-off dynos.

Reply:
Do you want me to pull the code all of the way out of the Autoscalers despite it being "similar" to autoscaler code (in that it uses the same endpoints as the autoscaler)? Or do you just want to make it *possible* to config the One-of creator to be separable from the autoscaler?

Reply:
Or do you want me to take another attempt at making a "job runner API interface" class which is separate from the autoscaler logic?

Reply:
@prescod Oh, I'm okay with having one_off_build be a method on the autoscaler class -- that's a convenient place to put it for swapping the implementation between local and Heroku. I just meant that I think it should get the name of the app to use for one-off dynos from a new setting, not from the AUTOSCALERS setting, because it seems pretty arbitrary that you're pulling out the 2nd entry from AUTOSCALERS. (Also I think the values are dicts, not an app name.)